app/controllers/api/pages_controller.rb

```diff
@@ -37,6 +38,20 @@ def featured
     render :index, format: :json
   end
 
+  def actions
+    return head :forbidden if @page.secure?
+    query = if @page.default_hidden?
+              published_status = Action.publish_statuses['published']
+              "page_id = '#{@page.id}' AND publish_status = '#{published_status}'"
+            else
+              hidden_status = Action.publish_statuses['hidden']
+              "page_id = '#{@page.id}' AND publish_status != '#{hidden_status}'"
+            end
+    page_number = { page_number: params[:page_number], per_page: params[:per_page] }
+    hashes, headers, _paginator = ActionReader.new(query).run(**page_number)
+    render json: { actions: hashes, headers: headers }
+  end
```

Comment:
I know we've discussed this before, and we don't usually agree on this. But I strongly believe all this logic should go in a service class. 

I'm not sure if the ActionReader class is used in other places, but you could extend the ActionReader class to handle all this, or build a new service class.